model/jpa/src/main/java/org/keycloak/models/jpa/entities/OrganizationEntity.java

```diff
@@ -35,7 +35,11 @@
 @Table(name="ORG")
 @Entity
 @NamedQueries({
-        @NamedQuery(name="getByRealm", query="select o from OrganizationEntity o where o.realmId = :realmId")
+        @NamedQuery(name="getByRealm", query="select o from OrganizationEntity o where o.realmId = :realmId order by o.name ASC"),
+        @NamedQuery(name="getByNameOrDomain", query="select distinct o from OrganizationEntity o inner join OrganizationDomainEntity d ON o.id = d.organization.id" +
+                " where o.realmId = :realmId AND (o.name = :search OR d.name = :search) order by o.name ASC"),
+        @NamedQuery(name="getByNameOrDomainContained", query="select distinct o from OrganizationEntity o inner join OrganizationDomainEntity d ON o.id = d.organization.id" +
+                " where o.realmId = :realmId AND (lower(o.name) like concat('%',:search,'%') OR d.name like concat('%',:search,'%')) order by o.name ASC")
```

Comment:
@sguilhen I can see you've used inner join which seems correct now as there is an assumption that organization has to have at least one domain set. I just want to highlight it ... in case it'll be chosen to allow orgs without domain this will have to be changed to left join, correct?

Reply:
@vramik yup.. I'm just not sure we will be doing that or if that was just an idea we were freely exploring without an actual requirement to have a domain-less org. For now I would keep the inner join.